instrumentation/opentelemetry-instrumentation-tortoiseorm/src/opentelemetry/instrumentation/tortoiseorm/__init__.py

```diff
@@ -0,0 +1,214 @@
+# limitations under the License.
+
+"""
+Instrument `tortoise-orm`_ to report SQL queries.
+Usage
+-----
+.. code:: python
+    from opentelemetry.instrumentation.tortoiseorm import TortoiseORMInstrumentor
+    from tortoise.contrib.fastapi import register_tortoise
+
+    register_tortoise(
+        app,
+        db_url=settings.db_url,
+        modules={"models": ["example_app.db_models"]},
+        generate_schemas=True,
+        add_exception_handlers=True,
+    )
+
+    TortoiseORMInstrumentor().instrument(tracer_provider=tracer)
+API
+---
+"""
+from typing import Collection
+
+try:
+    import tortoise.backends.asyncpg.client
+
+    TORTOISE_POSTGRES_SUPPORT = True
+except ModuleNotFoundError:
+    TORTOISE_POSTGRES_SUPPORT = False
+
+try:
+    import tortoise.backends.mysql.client
+
+    TORTOISE_MYSQL_SUPPORT = True
+except ModuleNotFoundError:
+    TORTOISE_MYSQL_SUPPORT = False
+
+try:
+    import tortoise.backends.sqlite.client
+
+    TORTOISE_SQLITE_SUPPORT = True
+except ModuleNotFoundError:
+    TORTOISE_SQLITE_SUPPORT = False
+
+import wrapt
+
+from opentelemetry import trace
+from opentelemetry.instrumentation.instrumentor import BaseInstrumentor
+from opentelemetry.instrumentation.tortoiseorm.package import _instruments
+from opentelemetry.instrumentation.tortoiseorm.version import __version__
+from opentelemetry.instrumentation.utils import unwrap
+from opentelemetry.semconv.trace import DbSystemValues, SpanAttributes
+from opentelemetry.trace import SpanKind
+from opentelemetry.trace.status import Status, StatusCode
+
+
+def _hydrate_span_from_args(connection, query, parameters) -> dict:
+    """Get network and database attributes from connection."""
+    span_attributes = {}
+    capabilities = getattr(connection, "capabilities", None)
+    if capabilities:
+        if capabilities.dialect == "sqlite":
+            span_attributes[SpanAttributes.DB_SYSTEM] = DbSystemValues.SQLITE.value
+        elif capabilities.dialect == "postgres":
+            span_attributes[SpanAttributes.DB_SYSTEM] = DbSystemValues.POSTGRESQL.value
+        elif capabilities.dialect == "mysql":
+            span_attributes[SpanAttributes.DB_SYSTEM] = DbSystemValues.MYSQL.value
+    dbname = getattr(connection, "filename", None)
+    if dbname:
+        span_attributes[SpanAttributes.DB_NAME] = dbname
+    dbname = getattr(connection, "database", None)
+    if dbname:
+        span_attributes[SpanAttributes.DB_NAME] = dbname
+    if query is not None:
+        span_attributes[SpanAttributes.DB_STATEMENT] = query
+    user = getattr(connection, "user", None)
+    if user:
+        span_attributes[SpanAttributes.DB_USER] = user
+    host = getattr(connection, "host", None)
+    if host:
+        span_attributes[SpanAttributes.NET_PEER_NAME] = host
+    port = getattr(connection, "port", None)
+    if port:
+        span_attributes[SpanAttributes.NET_PEER_PORT] = port
+
+    if parameters is not None and len(parameters) > 0:
```

Comment:
I think this should not be captured by default to prevent PII from leaking. Other instrumentations allow users to turn on this feature: https://opentelemetry-python-contrib.readthedocs.io/en/latest/instrumentation/dbapi/dbapi.html#api